pkg/proxy/metaproxier/meta_proxier.go

```diff
@@ -62,7 +62,8 @@ func (proxier *metaProxier) SyncLoop() {
 
 // OnServiceAdd is called whenever creation of new service object is observed.
 func (proxier *metaProxier) OnServiceAdd(service *v1.Service) {
-	if *(service.Spec.IPFamily) == v1.IPv4Protocol {
+	family, _ := servicesIPFamily(service)
```

Comment:
I guess that if err != nil or family = nil we should just simple return ... and log the error maybe?

Reply:
I can add logging if err != nil.
Assuming ipv4 is always supported, OnServiceAdd should fall back to ipv4 when err != nil or family = nil

What do you think ?

Reply:
I think that if you are not able to get an IP for the service doesn't matter to default to anything, kube-proxy won't need to do anything

Reply:
@aojea 
Please take another look.